### PR TITLE
Add more built-in rich text formats

### PR DESCRIPTION
**Pull request by arifin4web**
As discussed in  #4562, we are adding few more additional feature formats/identifier within wagtail so that it become more easier add any/all of these features with our rich text editor. Newly added Identifiers are:

- blockquote

- code
- superscript
- subscript
- strikethrough

Screen Shot 2019-03-14 at 10 48 36

### CODE REVIEW

File: wagtail/admin/wagtail_hooks.py [hunk 1]
```diff
@@ -459,6 +459,36 @@ def register_core_features(features):
...
+        })
+    )
+    features.register_converter_rule('contentstate', 'code-block', {
+        'from_database_format': {
+            'code': BlockElementHandler('code-block'),
+        },
+        'to_database_format': {
+            'block_map': {'code-block': 'code'}
+        }
+    })
```
Comment (thibaudcolas) [inline wagtail/admin/wagtail_hooks.py]: We might need to have a look at whether `code-block` can be included or not. I wouldn't be surprised if there things that made it incompatible with Wagtail's rich text processing.

File: client/src/components/Draftail/Draftail.scss [hunk 2]
```diff
@@ -103,3 +103,21 @@ $draftail-editor-font-family: $font-serif;
...
+        box-shadow: none;
+    }
+}
+
+.Draftail-block--blockquote {
+    border-left: 0.25em solid $color-grey-3;
+    color: $color-grey-2;
+    margin: 1em 0;
+    padding: 1em 2em;
+}
```
Comment (thibaudcolas) [inline client/src/components/Draftail/Draftail.scss]: 👍 could you add a screenshot of what this looks like to the PR?

File: wagtail/admin/wagtail_hooks.py [hunk 3]
```diff
@@ -518,3 +548,48 @@ def register_core_features(features):
...
+        })
+    )
+    features.register_converter_rule('contentstate', 'strikethrough', {
+        'from_database_format': {
+            's': InlineStyleElementHandler('STRIKETHROUGH'),
+        },
+        'to_database_format': {
+            'style_map': {'STRIKETHROUGH': 's'}
+        }
+    })
```
Comment (thibaudcolas) [inline wagtail/admin/wagtail_hooks.py]: I think this is missing inline `code` formatting?

Comment (arifin4web) [reply wagtail/admin/wagtail_hooks.py]: Do you mean a new identifier called `inline-code`?

File: wagtail/admin/wagtail_hooks.py (hunk 1 above)
Comment (arifin4web) [reply wagtail/admin/wagtail_hooks.py]: Dropping `code-block` for now. We can look into this later.

**Comment by thibaudcolas** [comment]
Merged! Thank you @arifin4web 🙂 this will be part of Wagtail v2.5.

**Pull request merged**